### PR TITLE
Use default dtype in `F.n_step_rnn` and `F.n_step_birnn`

### PR DESCRIPTION
**Pull request by takagi**
Merge cupy/cupy#1471  for appropriate cuDNN RNN support, and #5143 first

This PR is a part of #4582, makes `F.n_step_rnn` support the default dtype.

### CODE REVIEW

**Comment by takagi** [comment]
jenkins, test this please.

**Comment by takagi** [comment]
Merge #5143 first as well as cupy/cupy#1471

**Comment by takagi** [comment]
Jenkins, test this please.

File: chainer/functions/connection/n_step_rnn.py [hunk 1]
```diff
@@ -50,6 +50,19 @@ def _make_tensor_descriptor_array(xs):
     return PointerArray([d.value for d in descs], descs)
 
 
+def _get_cudnn_data_type(dtype):
```
Comment (kmaehashi) [inline chainer/functions/connection/n_step_rnn.py]: Should we redefine `cupy.cudnn.get_data_type` here?

Comment (takagi) [reply chainer/functions/connection/n_step_rnn.py]: It would be enough, I will fix it!

Comment (takagi) [reply chainer/functions/connection/n_step_rnn.py]: Fixed!

**Comment by takagi** [comment]
Jenkins, test this please.

**Comment by takagi** [comment]
Jenkins, test this please.

**Comment by takagi** [comment]
@kmaehashi Would you check this?

**Comment by takagi** [comment]
I close this PR and will implement based on the current master again.

**Comment by takagi** [comment]
I continue to change based on this branch.

**Pull request closed without merging**